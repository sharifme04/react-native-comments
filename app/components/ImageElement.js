import React from 'react';
import {
  StyleSheet,
  View,
  Image,
  TouchableWithoutFeedback,
  Dimensions,
  Text
} from 'react-native';
import PropTypes from "prop-types";
import Icon from 'react-native-vector-icons/FontAwesome';

const ImageElement = ({imageUrl, handleVisible, id,totalComments, handleImage}) => {
  const commentLength = id &&  totalComments && totalComments?.filter(c => c.imageId === id)?.length;
  return (
    <TouchableWithoutFeedback
      onPress={() => {
        handleVisible(true);
        handleImage(imageUrl, id);
      }}>
      <View style={styles.imageWrap}>
        <Image
          style={styles.image}
          source={{
            uri: imageUrl,
          }}
        />
        <View style={styles.thumbnailComment} >
        <Icon name="comment" size={15} color="#900" />
        {commentLength ? <Text>{commentLength}</Text>: null}
        </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

const styles = StyleSheet.create({
  imageWrap: {
    margin: 2,
    padding: 2,
    height: Dimensions.get('window').height / 4 - 8,
    width: Dimensions.get('window').width / 2 - 4,
    borderWidth: 1,
    borderColor: 'thistle',
  },
  image: {
    flex: 1,
    width: null,
    alignSelf: 'stretch',
  },
  thumbnailComment: {
    flexDirection: 'row',
  }
});

ImageElement.propTypes = {
  imageUrl: PropTypes.string,
  handleVisible:PropTypes.func,
  id: PropTypes.any,
  totalComments: PropTypes.array,
  handleImage: PropTypes.func
};

export default ImageElement;
