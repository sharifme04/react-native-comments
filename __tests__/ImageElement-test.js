import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ImageElement from '../app/components/ImageElement';

test('ImageElement ', () => {
  const tree = renderer.create(<ImageElement />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ImageElement empty images ', () => {
  const tree = renderer.create(<ImageElement imageUrl={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ImageElement with images ', () => {
  const tree = renderer
    .create(
      <ImageElement
        handleVisible={() => {}}
        handleImage={() => {}}
        key={4343}
        imageUrl={
          'https://images.pexels.com/photos/3408744/pexels-photo-3408744.jpeg'
        }
        id={37823}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
