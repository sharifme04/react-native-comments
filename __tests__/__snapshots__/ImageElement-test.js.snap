// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageElement  1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "thistle",
      "borderWidth": 1,
      "height": 325.5,
      "margin": 2,
      "padding": 2,
      "width": 371,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "width": null,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#900",
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`ImageElement empty images  1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "thistle",
      "borderWidth": 1,
      "height": 325.5,
      "margin": 2,
      "padding": 2,
      "width": 371,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "width": null,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#900",
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`ImageElement with images  1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "thistle",
      "borderWidth": 1,
      "height": 325.5,
      "margin": 2,
      "padding": 2,
      "width": 371,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://images.pexels.com/photos/3408744/pexels-photo-3408744.jpeg",
      }
    }
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
        "width": null,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#900",
            "fontSize": 15,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
